universum/lib/gravity.py

```diff
@@ -7,28 +9,28 @@
 
 
 class ModuleSettings:
-    def __init__(self, cls, main_settings):
+    def __init__(self, cls: Type['Module'], main_settings: 'HasModulesMapping') -> None:
         object.__setattr__(self, "cls", cls)
         object.__setattr__(self, "main_settings", main_settings)
 
-    def __getattribute__(self, item):
-        cls = object.__getattribute__(self, "cls")
+    def __getattribute__(self, item: str) -> Union[str, List[str]]:
+        cls: Type['Module'] = object.__getattribute__(self, "cls")
         main_settings = object.__getattribute__(self, "main_settings")
         for entry in cls.__mro__:
             try:
-                settings = getattr(main_settings, entry.__name__)
+                settings: 'HasModulesMapping' = getattr(main_settings, entry.__name__)
```

Comment:
I am sorry to say this, but while this member is `ModuleNamespace`, it doesn't in fact contain `active_modules`.
In general, I am not sure what benefits we can have from annotating it.
This is because it is designed to be dynamically populated from the command line arguments.
Probably, it can be fixed if we change `define_arguments` to static typing.